src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -2063,6 +2063,45 @@ closeGraphContext(unsigned int graphId)
   return ret ? -errno : ret;
 }
 
+void
+shim::
+open_graph_context(const zynqaie::hwctx_object* hwctx, const uuid_t xclbinId, unsigned int graph_id, xrt::graph::access_mode am)
+{
+  if (!hw_context_enable){
+    // for legacy flow
+    if (auto ret = openGraphContext(xclbinId, graph_id, am))
+      throw xrt_core::error("Failed to open graph context");
+    return;
+  }
+  // this is for multi slot case
+  auto shared = (hwctx->get_mode() != xrt::hw_context::access_mode::exclusive);
+  unsigned int flags = shared ? ZOCL_CTX_SHARED : ZOCL_CTX_EXCLUSIVE;
+  drm_zocl_open_graph_ctx graph_ctx = {};
+  graph_ctx.hw_context = hwctx->get_slotidx();
+  graph_ctx.flags = flags;
+  graph_ctx.graph_id = graph_id;
+  if (auto ret = ioctl(mKernelFD, DRM_IOCTL_ZOCL_OPEN_GRAPH_CTX, &graph_ctx))
+    throw xrt_core::error("Failed to open graph context hwctx shim");
+}
+
+void
+shim::
+close_graph_context(const zynqaie::hwctx_object* hwctx, unsigned int graph_id)
+{
+  if (!hw_context_enable) {
+    // for legacy flow
+    if (auto ret = closeGraphContext(graph_id))
+      throw xrt_core::error("Failed to close graph context");
+    return;
+  }
+  // this is for multi slot case
+  drm_zocl_close_graph_ctx graph_ctx = {};
+  graph_ctx.hw_context = hwctx->get_slotidx();
+  graph_ctx.graph_id = graph_id;
+  if (auto ret = ioctl(mKernelFD, DRM_IOCTL_ZOCL_CLOSE_GRAPH_CTX, &graph_ctx))
+    throw xrt_core::error("Failed to close graph context hwctx shim");
```

Comment:
Spacing and if you don't use the `ret` value, then don't save it.

```suggestion
  if (!hw_context_enable) {
    // for legacy flow
    if (closeGraphContext(graph_id))
      throw xrt_core::error("Failed to close graph context");
      
    return;
  }
  // this is for multi slot case
  drm_zocl_close_graph_ctx graph_ctx = {};
  graph_ctx.hw_context = hwctx->get_slotidx();
  graph_ctx.graph_id = graph_id;
  if (ioctl(mKernelFD, DRM_IOCTL_ZOCL_CLOSE_GRAPH_CTX, &graph_ctx))
    throw xrt_core::error("Failed to close graph context hwctx shim");
```